### ISSUE TITLE
Before Release of `pre-commit` Addition

### ISSUE THREAD

**Issue by mindlessroman** [tbroadley/spellchecker-cli#116]
I realized that the README will probably need an update about which `rev` to use.

```yaml
- repo: https://github.com/tbroadley/spellchecker-cli
  rev: <INSERT RELEASE HERE>
  hooks:
    - id: spellchecker-cli
      name: spellcheck
      types: [markdown]
      stages: # optional: if you want to specify stages to run the hook on
        - '' # see https://pre-commit.com/#confining-hooks-to-run-at-certain-stages
```
In the mean time, my team will use the commit hash, but just for awareness, @tbroadley.

**Comment by psychemedia** [comment]
For some reason I am getting an ` InvalidManifestError` for the following `pre-commit-config.yaml` file. Any hints as to what's wrong (is there a `pre-commit-config.yaml` validator anywhere?)

```yaml
repos:
  - repo: https://github.com/tbroadley/spellchecker-cli
    rev: v6.1.1
    hooks:
      - id: spellchecker-cli
        name: spellchecker
        description: A pre-commit implementation for spellchecker-cli
        entry: spellchecker
        language: node
        types: [markdown]
```

**Comment by tbroadley** [comment]
Looking at this part of the pre-commit documentation: https://pre-commit.com/#pre-commit-configyaml---hooks

I don't think the `hooks` object in a `.pre-commit-config.yaml` can have `description`, `entry`, or `language` properties.

We should adjust the example in this project's README.

**Comment by psychemedia** [comment]
I'm successfully using:

```yaml
  - repo: https://github.com/markdownlint/markdownlint
    rev: v0.12.0
    hooks:
      - id: markdownlint
        name: Markdownlint
        description: Run markdownlint on your Markdown files
        entry: mdl
        language: ruby
        files: \.(md|mdown|markdown)$
```

Are those fields perhaps defined as part of a particular pre-commit action?

I can't find any combination of keys (yet!), or even a minimal combination (yet!) that will run for me with `spellcheker-cli`.

**Comment by psychemedia** [comment]
FWIW, I am using `pre-commit` version: 3.4.0

**Comment by mindlessroman** [comment]
@psychemedia - I've successfully used it referencing the commit hash. Since the release you picked doesn't have a `pre-commit-config.yaml`, I don't think it can successfully run. 

```yaml
-   repo: https://github.com/tbroadley/spellchecker-cli
    rev: e4879b5 # TODO: Release tag
    hooks:
      - id: spellchecker-cli
        name: spellcheck
        types: [markdown]
        stages:
          - "pre-commit"
```

@tbroadley I think at the least, in the mean time having the `README.md` point to the commit hash until there's a release will probably help with this. I can make a PR for this?

**Comment by psychemedia** [comment]
@mindlessroman ah - that seems to do the trick... thanks...

**Comment by psychemedia** [comment]
Is there a way of specifying dictionaries in the hook config?

**Comment by psychemedia** [comment]
If I try to pass args from my local `.pre-commit-config,yaml` file, such as the following (or variants thereof):
 
`args: ['--no-suggestions', '--dictionaries', 'dictionary.txt'] I get an error:`

I get an error:

```text
spellchecker.............................................................Failed
- hook id: spellchecker-cli
- exit code: 1

A list of files is required.

```

Should the `pass_filenames` attribute be set as false in the hook declaration in the repo?

**Comment by mindlessroman** [comment]
I can try to take a stab at this shortly since I introduced it XD

**Comment by mindlessroman** [comment]
> Is there a way of specifying dictionaries in the hook config?

@psychemedia You can [specify the dictionaries](https://github.com/tbroadley/spellchecker-cli#configuration-files) in the `spellcheckerrc.[*]` file (my team's is in a `yaml` file):

```yaml
files:
  - "**/*.md"
  - "!SUPPORT.md"
  - "!CODE_OF_CONDUCT.md"
  - "!SECURITY.md"
dictionaries:
  - dictionary.txt
```
The precommit _should_ pick up these settings? But I can take a peek about making it so it can be handled in the pre-commit hook as well

**Comment by mindlessroman** [comment]
@tbroadley - did the pre-commit make it into the new release?

**Comment by tbroadley** [comment]
No, I never cut a new NPM release after #115 merged. I figured that it wasn't necessary because pre-commit would install the repo from GitHub, not NPM. But I can cut a new minor release just in case.

**Comment by tbroadley** [comment]
Just published 6.2.0.

**Comment by mindlessroman** [comment]
> Just published 6.2.0.

Awesome! I believe it would need to have the version tag that included the `.pre-commit-hooks.yaml` file.
